tools/debugger/debugger_core.py

```diff
@@ -930,7 +918,7 @@ def _parse_source(self, data):
             logging.debug("Pending breakpoints available")
             bp_list = self.pending_breakpoint_list
 
-            for breakpoint_index, breakpoint in bp_list.items():
+            for breakpoint_index, breakpoint in list(bp_list.items()):
```

Comment:
with a while loop this could be optimized

Reply:
It's not possible